garage/tf/policies/gaussian_mlp_policy_with_model.py

```diff
@@ -0,0 +1,158 @@
+            std_output_nonlinearity=std_output_nonlinearity,
+            std_parameterization=std_parameterization)
+
+        with tf.variable_scope(name, reuse=False):
+            _, self._mean_var, self._log_std_var, _, self._dist = \
+                self.model.build(state_input, dist)
+
+            self._f_dist = lambda x: tf.get_default_session().run(
+                [self._mean_var, self._log_std_var],
+                feed_dict={state_input: x})
+
+    @property
+    def vectorized(self):
+        """Vectorized or not."""
+        return True
+
+    def dist_info_sym(self,
+                      obs_var,
+                      state_info_vars=None,
+                      name='dist_info_sym'):
+        """Symbolic graph of the distribution."""
+        with tf.variable_scope(
+                self.name, reuse=True, auxiliary_name_scope=False):
+            _, mean_var, log_std_var, _, _ = self.model.build(
+                obs_var, DiagonalGaussian, name=name)
+        return dict(mean=mean_var, log_std=log_std_var)
+
+    def get_action(self, observation):
+        """Get action from the policy."""
+        flat_obs = self.observation_space.flatten(observation)
+        mean, log_std = [x[0] for x in self._f_dist([flat_obs])]
+        rnd = np.random.normal(size=mean.shape)
+        action = rnd * np.exp(log_std) + mean
+        return action, dict(mean=mean, log_std=log_std)
+
+    def get_actions(self, observations):
+        """Get actions from the policy."""
+        flat_obs = self.observation_space.flatten_n(observations)
+        means, log_stds = self._f_dist(flat_obs)
+        rnd = np.random.normal(size=means.shape)
+        actions = rnd * np.exp(log_stds) + means
+        return actions, dict(mean=means, log_std=log_stds)
+
+    def get_reparam_action_sym(self,
+                               obs_var,
+                               action_var,
+                               old_dist_info_vars,
+                               name="get_reparam_action_sym"):
+        """
+        Get symbolically reparamterzied action represnetation.
+
+        Given observations, old actions, and distribution of old actions,
+        return a symbolically reparameterized representation of the actions in
+        terms of the policy parameters.
+
+        :param obs_var:
+        :param action_var:
+        :param old_dist_info_vars:
+        :return:
+        """
+        new_dist_info_vars = self.dist_info_sym(obs_var, name=name)
+        new_mean_var, new_log_std_var = new_dist_info_vars[
+            "mean"], new_dist_info_vars["log_std"]
+        old_mean_var, old_log_std_var = old_dist_info_vars[
+            "mean"], old_dist_info_vars["log_std"]
+        epsilon_var = (action_var - old_mean_var) / (
+            tf.exp(old_log_std_var) + 1e-8)
+        new_action_var = new_mean_var + epsilon_var * tf.exp(new_log_std_var)
+        return new_action_var
+
+    def log_diagnostics(self, paths):
```

Comment:
you can remove this -- it never gets called

Reply:
I saw it is called in `OffPolicyRLAlgorithm.log_diagnostics`.
```python
    def log_diagnostics(self, paths):
        """Log diagnostic information on current paths."""
        self.policy.log_diagnostics(paths)
        self.qf.log_diagnostics(paths)
```

Reply:
(double check)